crates/sim2h/src/lib.rs

```diff
@@ -197,56 +182,42 @@ pub enum DhtAlgorithm {
     NaiveSharding { redundant_count: u64 },
 }
 
-mod sim2h_state;
-pub(crate) use sim2h_state::*;
-
-#[derive(Debug)]
-struct Sim2hComHandleMessage {
-    uri: Lib3hUri,
-    message: WireMessage,
-    signer: AgentId,
-}
-
-#[derive(Debug)]
-struct Sim2hComHandleJoined {
-    uri: Lib3hUri,
-    space_address: SpaceHash,
-    agent_id: AgentId,
-    message: WireMessage,
-}
+#[allow(dead_code)]
+mod mono_ref;
+use mono_ref::*;
 
-#[derive(Debug)]
-enum Sim2hCom {
-    HandleMessage(Box<Sim2hComHandleMessage>),
-    HandleJoined(Box<Sim2hComHandleJoined>),
-    Disconnect(Vec<Lib3hUri>),
-}
+#[allow(dead_code)]
+mod sim2h_im_state;
 
 #[derive(Clone)]
 /// A clonable reference to our Sim2h instance that can be passed
 /// into `'static` async blocks && still be able to make sim2h calls
 struct Sim2hHandle {
-    state: Arc<tokio::sync::Mutex<Sim2hState>>,
-    send_com: tokio::sync::mpsc::UnboundedSender<Sim2hCom>,
+    im_state: sim2h_im_state::StoreHandle,
```

Comment:
is there a reason to call this `im_state` instead of just `state`?

Reply:
lol, while doing the transition I had both of them in there... I can rename it now.